### PR TITLE
Add overloads for `asyncio.sleep()`.

### LINKED ISSUE #7866: `asyncio.sleep()` should have an overload
This is from https://github.com/microsoft/pyright/issues/3475.

With the latest version of pyright (1.1.247), this code results in an error:

```py
from typing import Awaitable, TypeVar
import asyncio
T = TypeVar("T")

async def wrap(coro: Awaitable[T]) -> T:
    return await coro

async def wrap_sleep() -> None:
    await wrap(asyncio.sleep(0))
```

The result when I run `pyright` on it:

```
Found 1 source file
/..../test/test.py
  /..../test/test.py:9:16 - error: Argument type is partially unknown
    Argument corresponds to parameter "coro" in function "wrap"
    Argument type is "Coroutine[Any, Any, Unknown]" (reportUnknownArgumentType)
1 error, 0 warnings, 0 informations 
```

The fix, as suggested by @JelleZijlstra and @erictraut, is to add an overload like this:

```py
 async def sleep(delay: float) -> None: ...
```

### PR DESCRIPTION
**Pull request by wch**
Closes #7866. This adds an overload to `asyncio.sleep()`, so that when it is called _without_ `return=None`, the type checker knows that the return type is `None` instead of `unknown`.

Also related to https://github.com/microsoft/pyright/issues/3475.

### CODE REVIEW

File: stdlib/asyncio/tasks.pyi [hunk 1]
```diff
@@ -270,6 +270,9 @@ def run_coroutine_threadsafe(coro: _FutureT[_T], loop: AbstractEventLoop) -> con
 
 if sys.version_info >= (3, 10):
     def shield(arg: _FutureT[_T]) -> Future[_T]: ...
+    @overload
+    async def sleep(delay: float) -> None: ...
+    @overload
     async def sleep(delay: float, result: _T = ...) -> _T: ...
```
Comment (JelleZijlstra) [inline stdlib/asyncio/tasks.pyi:276]:
```suggestion
    async def sleep(delay: float, result: _T) -> _T: ...
```
So the overloads don't overlap.

File: stdlib/asyncio/tasks.pyi [hunk 2]
```diff
@@ -281,6 +284,9 @@ if sys.version_info >= (3, 10):
 
 else:
     def shield(arg: _FutureT[_T], *, loop: AbstractEventLoop | None = ...) -> Future[_T]: ...
+    @overload
+    async def sleep(delay: float, *, loop: AbstractEventLoop | None = ...) -> None: ...
+    @overload
     async def sleep(delay: float, result: _T = ..., *, loop: AbstractEventLoop | None = ...) -> _T: ...
```
Comment (JelleZijlstra) [inline stdlib/asyncio/tasks.pyi:290]:
```suggestion
    async def sleep(delay: float, result: _T, *, loop: AbstractEventLoop | None = ...) -> _T: ...
```

**Comment by wch** [comment]
Looking for a bit of advice. This part fails the CI checks on Python <= 3.9. (Python >= 3.10 uses a different code path which seems to be fine.)

```py
    @overload
    async def sleep(delay: float, *, loop: AbstractEventLoop | None = ...) -> None: ...
    @overload
    async def sleep(delay: float, result: _T = ..., *, loop: AbstractEventLoop | None = ...) -> _T: ...
```

```
stdlib/asyncio/tasks.pyi:288: error: Overloaded function signatures 1 and 2 overlap with incompatible return types  [misc]
```

Is it possible to make this work?

**Comment by AlexWaygood** [comment]
> Is it possible to make this work?

Committing @JelleZijlstra's suggestions he made in review just now should do it :)

**Comment by wch** [comment]
@AlexWaygood Thanks, he must have posted that as I was typing. :)

File: stdlib/asyncio/tasks.pyi (hunk 1 above)
Comment (wch) [reply stdlib/asyncio/tasks.pyi:276]: This one doesn't seem to cause errors in CI. Should I still commit this change?

Comment (AlexWaygood) [reply stdlib/asyncio/tasks.pyi]: I just committed it for you :) I'm not sure why it's not causing errors in CI. I feel like it should do.

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**